etc/cmake/options.cmake

```diff
@@ -88,6 +88,15 @@ if (OTBR_SRP_ADVERTISING_PROXY)
     target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_SRP_ADVERTISING_PROXY=1)
 endif()
 
+option(OTBR_USE_OT_SRP_ADV_PROXY "Enable OT core Advertising Proxy" OFF)
```

Comment:
nit: we don't use the "_USE_" prefix in any other config macro. Should we rename to `OTBR_OT_SRP_ADV_PROXY`?